Update README, in preparation for GA release (#8)
mattermost/mattermost-plugin-welcomebot: README.md

# Welcome Bot Plugin ![CircleCI branch](https://img.shields.io/circleci/project/github/mattermost/mattermost-plugin-welcomebot/master.svg)

Use this plugin to improve onboarding and HR processes. It adds a Welcome Bot that helps add new team members to channels.

![image](https://user-images.githubusercontent.com/13119842/58736467-fd226400-83cb-11e9-827b-6bbe33d062ab.png)

*Welcome a new team member to Mattermost Contributors team. Then add the user to a set of channels based on their selection.*

![image](https://user-images.githubusercontent.com/13119842/58736540-30fd8980-83cc-11e9-8e8e-94ea3042b3b1.png)

## Configuration

1. Go to **System Console > Plugins > Management** and click **Enable** to enable the Welcome Bot plugin.
    - If you are running Mattermost v5.11 or earlier, you must first go to the [releases page of this GitHub repository](https://github.com/mattermost/mattermost-plugin-welcomebot/releases), download the latest release, and upload it to your Mattermost instance [following this documentation](https://docs.mattermost.com/administration/plugins.html#plugin-uploads).

> 2. Modify your `config.json` file to include your custom attributes, under the `PluginSettings`. See below for an example of what this should look like.

Reviewer: 1/5 `your custom attributes` is confusing. Maybe `your Welcome Bot's messages and actions`?

Author: Good catch, copy-paste error :)

## Usage

To configure the Welcome Bot, edit your `config.json` file with a message you want to send to a user in the following format:

```
"WelcomeMessages": [
    {
        "TeamName": "your-team-name",
        "DelayInSeconds": 3,
        "Message": [
            "Your welcome message here. Each list item specifies one line in the message text."
        ]
        "AttachmentMessage": [
            "Attachment message containing user actions"
        ],
        "Actions" : [
            {
                "ActionType": "button",
                "ActionDisplayName": "User Action",
                "ActionName": "action-name",
                "ActionSuccessfulMessage": [

>                     "Message posted after use takes this action and joins channels specified by 'ChannelsAddedTo'."

Reviewer: ```suggestion
                    "Message posted after the user takes this action and joins channels specified by 'ChannelsAddedTo'."
```

                ]
                "ChannelsAddedTo": ["channel1", "channel2"],
            },
            {
                "ActionType": "automatic",
                "ChannelsAddedTo": ["channel3", "channel4"]
            }
        ]
    }
]
```

where

> - **TeamName**: The team for which the Welcome Bot sends a message for. Refers to the team handle used in the URL. For example, in the following URLm the **TeamName** value is `myteam`: https://example.com/myteam/channels/mychannel

Reviewer: 👍 nice detailed examples

- **DelayInSeconds**: The number of seconds after joining a team that the user receives a welcome message.
- **Message**: The message posted to the user.
- (Optional) **AttachmentMessage**: Message text in attachment containing user action buttons. 
- (Optional) **Actions**: Use this to add new team members to channels automatically or based on which action button they pressed.

>     - **ActionType**: One of `button` or `automatic`. When a `button`, enables uses to select which types of channels they want to join. When `automatic`, the user is automatically added to the specified channels.

Reviewer: ```suggestion
   - **ActionType**: One of `button` or `automatic`. When `button`: enables uses to select which types of channels they want to join. When `automatic`: the user is automatically added to the specified channels.
```

    - **ActionDisplayName**: Sets the display name for the user action buttons.
    - **ActionName**: Sets the action name used by the plugin to identify which action is taken by a user.

>     - **ActionSuccessfulMessage**: Message posted after use takes this action and joins the specified channels.

Reviewer: ```suggestion
    - **ActionSuccessfulMessage**: Message posted after the user takes this action and joins the specified channels.
```

                        "ActionDisplayName": "Developer",
                        "ActionName": "developer-action",
                        "ChannelsAddedTo": ["bugs", "jira-tasks", "sprint-planning"],
                        "ActionSuccessfulMessage": [
                            "### Awesome! I have added you to the following developer channels:",
                            "~bugs - To help investigate or report bugs",
                            "~jira-tasks - To stay updated on Jira tasks",
                            "~sprint-planning - To plan and manage your team's Jira sprint"
                        ]
                    },
                    {
                        "ActionType": "button",
                        "ActionDisplayName": "Account Manager",
                        "ActionName": "account-manager-action",
                        "ChannelsAddedTo": ["leads", "sales-discussion", "win-loss-analysis"],
                        "ActionSuccessfulMessage": [
                            "### Awesome! I have added you to the following developer channels:",
                            "~leads - To stay updated on incoming leads",

>                             "~help-desk - To connect with your support team for quick Q&A

Reviewer: ```suggestion
```

                            "~sales-discussion - To collaborate with your fellow account managers",
                            "~win-loss-analysis - To conduct win-loss analysis of closed deals"
                        ]
                    },
                    {
                        "ActionType": "button",
                        "ActionDisplayName": "Support",
                        "ActionName": "support-action",
                        "ChannelsAddedTo": ["bugs", "customer-support", "leads"],
                        "ActionSuccessfulMessage": [
                            "### Awesome! I have added you to the following developer channels:",
                            "~bugs - To help investigate or report bugs",
                            "~customer-support - To troubleshoot and resolve customer issues",

>                             "~help-desk - To connect with account managers for quick Q&A"

Reviewer: ```suggestion
                            "~leads - To discuss potential accounts with other account managers"
```
